src/js/components/DateInput/DateInput.js

```diff
@@ -62,6 +64,20 @@ const DateInput = forwardRef(
       initialValue: defaultValue,
     });
 
+    let timestamp;
+    if (Array.isArray(defaultValue) && defaultValue.length) {
+      timestamp = getTimestamp(defaultValue[0]);
+    } else if (typeof defaultValue === 'string') {
+      timestamp = getTimestamp(defaultValue);
+    } else if (Array.isArray(value) && value.length) {
+      timestamp = getTimestamp(value[0]);
+    } else if (typeof value === 'string') {
+      timestamp = getTimestamp(value);
+    }
+
+    // normalize value based on timestamp vs user's local timezone
+    const adjustedValue = getAdjustedDate(value, timestamp);
```

Comment:
"date" would make this more apparent later on in the code that we are working with a date here.
adjustedValue --> adjustedDate or adjustedDateValue or normalizedDate